src/Components/NavBar/Menus/MobileNavMenu.tsx

```diff
@@ -1,144 +1,244 @@
+import React from "react"
+import styled from "styled-components"
+import { MenuData, MenuLinkData } from "../menuData"
+import { MobileLink } from "./MobileLink"
+import {
+  NavigatorContextProvider,
+  useNavigation,
+} from "./NavigatorContextProvider"
 
-import { AnalyticsSchema, SystemContext } from "Artsy"
-import { useTracking } from "Artsy/Analytics/useTracking"
-
-interface MobileNavMenuProps {
-  onNavItemClick?: () => void
+interface Props {
+  isOpen: boolean
+  menuData: MenuData
 }
 
-export const MobileNavMenu: React.FC<MobileNavMenuProps> = props => {
-  const { onNavItemClick } = props
-  const { trackEvent } = useTracking()
-  const { user, EXPERIMENTAL_APP_SHELL } = useContext(SystemContext)
-  const isLoggedIn = Boolean(user)
-
-  const trackClick = event => {
-    const link = event.target
-    const text = link.innerText
-    const href = link.parentNode.getAttribute("href")
-
-    trackEvent({
-      action_type: AnalyticsSchema.ActionType.Click,
-      subject: text,
-      destination_path: href,
-    })
-
-    // TODO: Remove after EXPERIMENTAL_APP_SHELL AB test ends.
-    if (EXPERIMENTAL_APP_SHELL && href === "/collect") {
-      onNavItemClick()
-    }
-  }
+export const MobileNavMenu: React.FC<Props> = props => {
+  const {
+    links: [artworks, artists],
+  } = props.menuData
+  const { user } = useSystemContext()
 
   return (
-    <MobileNavContainer py={1} flexDirection="column" onClick={trackClick}>
-      <MobileLink href="/collect">Artworks</MobileLink>
-      <MobileLink href="/auctions">Auctions</MobileLink>
-      <MobileLink href="/galleries">Galleries</MobileLink>
-      <MobileLink href="/fairs">Fairs</MobileLink>
-      <MobileLink href="/articles">Editorial</MobileLink>
-      <MobileLink href="/artists">Artists</MobileLink>
-      <MobileLink href="/shows">Shows</MobileLink>
-      <MobileLink href="/institutions">Museums</MobileLink>
-
-      <Box px={2}>
-        <Separator my={1} />
-      </Box>
-
-      {isLoggedIn ? (
-        <>
-          <MobileLink href="/works-for-you">Works for you</MobileLink>
-          <MobileLink href="/user/edit">Account</MobileLink>
-        </>
-      ) : (
-        <>
-          <MobileLink
-            href={`/log_in?intent=signup&trigger=click&contextModule=Header`}
-          >
-            Login
-          </MobileLink>
-          <MobileLink
-            href={`/sign_up?intent=signup&trigger=click&contextModule=Header`}
-          >
-            Sign up
-          </MobileLink>
-        </>
-      )}
-    </MobileNavContainer>
+    <NavigatorContextProvider>
+      <MenuViewport>
+        <AnimatingMenuWrapper isOpen={props.isOpen}>
+          <ul>
+            <MobileSubmenuLink menu={(artworks as MenuLinkData).menu}>
+              {(artworks as MenuLinkData).menu.title}
+            </MobileSubmenuLink>
+
+            <MobileSubmenuLink menu={(artists as MenuLinkData).menu}>
+              {(artists as MenuLinkData).menu.title}
```

Comment:
This double casting is a bit weird -- is it `menu` or is it `menu.title`? Necessary?

Reply:
Yeah this was kinda weird. `artworks` and `artists` represent [LinkData](https://github.com/artsy/reaction/blob/a5c1f60e51beb4665d60170c3f39a34cd11cd919/src/Components/NavBar/menuData.ts#L6) which is a union type of `MenuLinkData` and `SimpleLinkData` but TypeScript was only recognizing it as a SimpleLinkData type.